VogtObserver/Generator.cs

```diff
@@ -4,7 +4,34 @@
 
 namespace VogtObserver
 {
-    class Generator
+    class Generator : IObservable
     {
+
+          private List<IObserver> _observers;
+
+        public void Add(IObserver o)
+        {
+            _observers.Add(o);
+        }
+
+        public void Notify()
+        {
+            foreach (IObserver o in _observers)
+            {
+                o.Update();
+            }
+        }
+
+        public void Remove(IObserver o)
+        {
+            _observers.Remove(o);
+        }
+
+        public int GetRandomInt()
```

Comment:
You need to call Notify somewhere if this is something being observed.

This function is the function being observed. Therefore, this should look more like:
```
public int GetRandomInt() {
  Notify(random.Next(1,1000));
}
```

You could build this in a way that the newest sum is on the class, so the observer isnt parameterized, but it'd get a lot more complex to make it "right". You'd store each change as an "event" and the observers would have to know which event it looked at last.

Instead, parameterize the Notify() function so that you can notify the observers of the new random number!

Its kinda like this:
```
public class Airport extends Observable {

    private final String name;

    public Airport(String name) {
        this.name = name;
    }

    public void planeLanded(String flight) {
        setChanged();
        notifyObservers(flight);
    }

    @Override
    public String toString() {
        return name;
    }
}
```

Reply:
The notifyObservers(flight) is passing in the flight that landed.